### PR TITLE
VideoPress Onboarding: add support for an optional "more" button on Intro step

### PR DESCRIPTION
**Pull request by jgcaruso**
#### Proposed Changes

* Add support for an **_optional_** button that pops a modal on the Intro page for all onboarding flows
* adds VideoPress custom modal content

SCR-20230206-mj9

| desktop | mobile |
| --- | --- |
| SCR-20230206-mfe | SCR-20230206-mf6 |

NOTE: The `z-index` on the logo bar (`signup-header`) component is overridden to 0 on the `intro` page only to prevent it from overlapping the new modal. This change appears to work OK across all flows that use the `intro` flow (link-in-bio and newsletter). It isn't clear why this was initially set to `1` but it is `-1` in the original component, so presumably it was to fix an issue on one of the other steps.

cc @Automattic/vertex as it appears your team has modified these files recently.

#### Testing Instructions

* pull PR
* `yarn && yarn start`
* Visit `/setup/videopress`
* You should see the "Learn More" button
* Clicking it should open the modal
* Clicking `X` should close the modal
* In the modal, the links should all go somewhere
   *  Videomaker -> Videomaker theme page (jetpack because the wpcom one forces you to choose a site first)
   * Videopress -> videopress.com
   * Get Started -> next step in the flow??
   * "or learn more about videopress" -> videopress.com
* Test the "generic-ness" of the modal  by adding the following code snippet to the Newsletter flow in `intro/index.tsx` around line 47
```
modal: {
    buttonText: __( 'HI' ),
    content: null,
},
```
* then view the flow at `/setup/newsletter` -> the button should work as expected, but with no content
* Test the intro pages of other flows to ensure that the logo appears correctly
    * `/setup/link-in-bio`
    * `/setup/newsletter`

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx [hunk 1]
```diff
@@ -10,20 +13,62 @@ export interface IntroContent {
...
 
 const Intro: React.FC< Props > = ( { onSubmit, introContent } ) => {
+	const [ showModal, setShowModal ] = useState( false );
+	const { title, text, buttonText, modal } = introContent;
+
+	const modalClasses = classNames( 'intro__more-modal', {
+		show: showModal,
+	} );
+
+	const handleMoreClick = () => {
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx:37]: Can we fire a tracks event here to see what kind of usage this gets?

Comment (jgcaruso) [reply client/landing/stepper/declarative-flow/internals/steps-repository/intro/intro.tsx:37]: Added a new prop for providing a custom click handler here, and added our tracks event to it for the VideoPress flow `calypso_videopress_signup_learn_more_button_clicked`

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			</div>
+			<ul className="videopress-intro-modal__list">
+				<li>
+					<CheckmarkIcon />
+					<span>
+						{ translate( '{{a}}Videomaker{{/a}}, a premium theme optimized to display videos.', {
+							components: {
+								a: (
+									<a
+										href="https://jetpack.com/features/design/themes/showcase/videomaker/"
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx]: Maybe we should use this link instead: https://wordpress.org/themes/videomaker/

Might be less confusing than linking to Jetpack? Or here maybe? https://videomakerdemo.wordpress.com/

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+					<CheckmarkIcon />
+					<span>{ translate( 'Up to 200GB of storage and your own domain for a year.' ) }</span>
+				</li>
+			</ul>
+			<div className="videopress-intro-modal__button-column">
+				<Button className="intro__button" primary onClick={ onSubmit }>
+					{ translate( 'Get Started' ) }
+				</Button>
+				<Button plain className="learn-more">
+					{ translate( 'Or learn more about VideoPress.' ) }
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx]: This button doesn't do anything, is it meant to link to VideoPress.com?

**Review by roundhill — CHANGES_REQUESTED** [review]
Working fine besides a button that needs to link to videopress.com

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+				</Button>
+			</div>
+			<div className="videopress-intro-modal__screenshots">
+				<img
+					src="https://videopress2.files.wordpress.com/2023/02/videopress-modal-screenshots.png"
+					alt="Mobile device screenshot samples of the Videomaker theme."
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx]: Can `translate()` be used here to support localization?

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx (hunk 2 above)
Comment (jgcaruso) [reply client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx]: I think https://videomakerdemo.wordpress.com/ is better.

If you go to https://wordpress.org/themes/videomaker/ it makes you choose a site if you are already logged in. It isn't very useful which is why I chose the Jetpack link... but the theme demo makes more sense than jetpack.

File: client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx (hunk 3 above)
Comment (jgcaruso) [reply client/landing/stepper/declarative-flow/internals/steps-repository/intro/videopress-intro-modal-content.tsx]: I thought I did at one point, maybe didn't commit it 😅

**Review by roundhill — APPROVED** [review]
Changes look good! :shipit:

**Comment by evilluendas** [comment]
I've pushed a few small visual tweaks to the modal (so it's not as big in very large screens), and the animation. I think I haven't broken anything elsewhere, but please take a look just in case! 🙏 

I think the mobile version is usable but there are a couple of things I'd like to tweak. Perhaps we can deploy this and make those adjustments separately, so we can start gathering usage data asap?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7800692

Thank you @jgcaruso for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7800692).

**Pull request merged**